### ISSUE TITLE
Is this project dead?

### ISSUE THREAD

**Issue by mhinton** [kadirahq/flow-router#677]
There seem to be no commits for the last 4 months. What happened to version 4?  Is Kadira using react-router now instead?

**Comment by shyal** [comment]
It seems development is slowing down on kadira. But it's also time for everyone using these much needed repos to start contributing to them.

**Comment by abecks** [comment]
I think the organization responsible for Kadira/FlowRouter/FastRender is moving into React development (Storybook). They seem to be putting most of their manpower behind that. All of their Blaze-related packages are mostly abandoned.

**Comment by shyal** [comment]
What does FlowRouter have to do with Blaze though? It's a router, which has nothing to do with their template rendering engine.

My point is that the Meteor community is mostly filled with whiny little [insert expletive]. The whole point of open source is that when packages don't behave the way people want them to, they can fork them, fix the issue, create a pull request, have a few exchanges with the author during the review, and their issue is fixed.

Instead people are logging hundreds, and hundreds, and hundreds of issues. Guys and girls, I'll give you a hint, and it starts with "fork" and ends with "request".

If you love Meteor then you know what to do.. else give it a couple of years, and packages like this one with go into disrepair, and your beloved Meteor will fall out of favour.

Believe or not, you'll only have yourselves to blame.

/rant

**Comment by jamiter** [comment]
@shyal, there are pull requests open for this repository without any response. So welcome to the real world where package authors abandon packages and there are no "few exchanges with the author" or any review. Kadira moved on. To keep this package from dying somebody will have to be the new maintainer and convince everyone to start using the new package. I wish that person a lot of luck and endurance.

**Comment by shyal** [comment]
@jamiter thank you for your input.

> there are pull requests open for this repository without any response

:)

That's correct. Indeed that's a special case, I was speaking in general terms. If there are open pull requests, then either the existing repo collaborators can be granted the right to merge pull requests by the current repo owner, or someone's fork can become the new official repo, and we can ask Kadira to add a link to the new repository at the top of the README.md (as is the case with Meteor Up for example, although the new package is by the same people).

> To keep this package from dying somebody will have to be the new maintainer and convince everyone to start using the new package.

Exactly. For example I'm currently implementing a new deployment solution [meteor-station](https://github.com/shyal/meteor-station) in response to not being fussed with Docker (mupx) and mup no longer being maintained. I'm not far from being able to put it on NPM, and will make sure it's all nicely documented etc.

Similarly, someone needs to champion FlowRouter. I'm strongly considering writing my own router as well, to seamlessly store data in the URL à la TemplateVar, but via the Router. But there's only so much I'm willing to do in parallel.

I strongly encourage the Meteor Community at large to "scratch their own itch".

> I wish that person a lot of luck and endurance.

Well yes, because there are so many takers, and so few givers in highly accessible frameworks like ROR or Meteor. But nothing will change if no one gently points it out once in a while. ;)

Regards.

**Comment by NitroBAY** [comment]
500 issues, about 150 PR. It's not bad

**Comment by abecks** [comment]
The authors should update the readme stating that the package is abandoned and turn it over to someone else. Otherwise issues with subjects like this are all this repo is going to see from here on out. 

The frustrating thing is that they have completely gone into React related dev without any communication on these repos. Why should I use their new stuff only to wait for them to quickly abandon it again?

**Comment by shyal** [comment]
I've just sent an email to all 3 members of KadiraHQ. Let's see what say.

@abecks 

> Why should I use their new stuff only to wait for them to quickly abandon it again?

You need to shift your thinking from a "user" to a "developer" or "contributor". This is GitHub, not McDonald's.

**Comment by ForsakenHarmony** [comment]
@shyal any response?

**Comment by shyal** [comment]
Negative.

**Comment by ForsakenHarmony** [comment]
Any viable alternative?

**Comment by shyal** [comment]
I'm not sure. I'm posting a link to this thread in the Meteor IRC chan now.

**Comment by shyal** [comment]
https://forums.meteor.com/t/flow-router-is-dead-long-live-flow-router/28861

**Comment by NitroBAY** [comment]
@shyal you always can fork flow-router as seen of the state of this package I think devs would support you if you ask to those who PR to PR to your new project (or juste merge with them).

**Comment by shyal** [comment]
@NitroBAY I'd be happy to. In fact: https://github.com/shyal/flow-router I'm still puzzled by the original author's complete radio silence.

**Comment by shyal** [comment]
@NitroBAY 

See in fact, the package was originally [meteorhacks:flow-router](https://atmospherejs.com/meteorhacks/flow-router), but as you can see they updated the readme. The link to the github repo was also changed to Kadira's.

But with no response from kadira on this topic, it will hard to organise a handoff without me having to go around convincing people to use the new package, while confusion reigns, thing which I'm not really willing to do.

**Comment by ForsakenHarmony** [comment]
The official guide uses FlowRouter, so it's hard indeed

**Comment by sahanDissanayake** [comment]
Arunoda was the go to guy. But seems like he is now involved heavily on react-storybook
https://github.com/arunoda?tab=activity

**Comment by shyal** [comment]
@sahanDissanayake thanks. We're aware of these things. If you like, you can respond directly on the meteor forums thread, as we're going around in circles in this issue (and by that I mean that Arunoda, meteor-storybook etc. were already discussed here).

**Comment by ForsakenHarmony** [comment]
maybe an issue on storybook, even though unrelated, might bring the needed attention to this

**Comment by onedr0p** [comment]
@kadirahq appears to be updating all their other repos. I wish they would state their feelings on this package (whether for good or bad), it would be nice to know. I am switching to react-router I guess.

**Comment by abecks** [comment]
Still no response from @arunoda ? I can't wait for when Storybook is just as abandoned as the rest of their packages.

**Comment by joarobles** [comment]
investing here too @abecks

**Comment by shyal** [comment]
Kadira has been one of the top developers in the meteor community. I'm pretty sure storybook will really be something to look forward to.

**Comment by ForsakenHarmony** [comment]
@shyal wanna join my org, better than a fork by a single person https://github.com/meteorcom

**Comment by aadamsx** [comment]
I've recreated this project under the name FineRouter: https://github.com/aadamsx/fine-router

I'm in the process of converting the readme and some code over now.  PRs and contributors are very welcome.  I'll post something in the forums when I've got the conversion complete.

**Comment by julian-rare** [comment]
@aadamsx does it support POST, PUT routing? right now I'm thinking of dropping flow-router for Picker because of this. (customer restriction that says app must work even wen js disabled)

**Comment by aadamsx** [comment]
@julian-rare Picker is for server side routing only.  FineRouter aka FlowRouter is client side routing only.

The two are not mutually exclusive, in fact they work very well together.

**Comment by aadamsx** [comment]
I'll try to answer any questions I can over on the FineRouter repo.  FineRouter is FlowRouter as of today, but it will accept PRs and contributor roles for anyone interested.  I can't say I'll be able to solve any issues, but I'll try to respond.

**Comment by elie222** [comment]
Any updates here?

**Comment by ForsakenHarmony** [comment]
I don't think it's a good idea to have everyone fork it working on it on their own

**Comment by aadamsx** [comment]
> I don't think it's a good idea to have everyone fork it working on it on their own

You're right, I didn't fork FlowRouter, I spun it off into a open repository called FineRouter.  I kept the commit history that was in place here and anyone that would like to PR or get contributor status can.

**Comment by NitroBAY** [comment]
Very disappointed by the non-reaction of Kadira, Meteor is tightly bind to its packages (as the guide IS opinionated and Meteor lacks some basic functionality such as router) so the fact that such an important society as Kadira drop support without telling anyone of the officially recommended router is a shame, it really weakens all the little Meteor's ecosystem.

**Comment by julian-rare** [comment]
I've migrated to react-router with  https://github.com/thereactivestack/meteor-react-router-ssr

**Comment by rezaxdi** [comment]
As @arunoda mentioned in his blog post https://voice.kadira.io/its-time-thank-you-bye-meteor-115dd649a8#.59qpcurxy , now everyone interested in continuing this router (specially those who have made a fork and maintaining flowrouter there) can request for transition.

**Comment by livimonte** [comment]
This is a great option of @VeliovGroup: https://github.com/VeliovGroup/flow-router

**Comment by cyclops24** [comment]
Guys please join us to talk about FlowRouter Alternative:
[What a best alternative for FlowRouter now after lost arunoda and kadira?](https://forums.meteor.com/t/what-a-best-alternative-for-flowrouter-now-after-lost-arunoda-and-kadira/31846/33)

**Comment by rezaxdi** [comment]
FlowRouter isn't dead, just check the reply above you.

**Comment by julian-rare** [comment]
in my case moved to react-router with server side rendering

**Comment by abecks** [comment]
So whos fork are we going with? @VeliovGroup or @aadamsx ?

I have taken up `fast-render` and `inject-data` and fixed them for 1.4.2+, see my repositories.

**Comment by bensventures** [comment]
Didn't realise some forks were continuing the work, so I created another fork of FlowRouter, specifically the SSR branch, to add support for 1.4.2, using @abecks' packages. 

This is probably very badly hacked but it works for me needs so far: https://github.com/bensventures/flow-router

FlowRouter should not be left to die. It is pretty stable, is used in a lot of projects and it everywhere in the official documentation. And just for the sake of choice, we should keep it around along with other alternatives like react-router.

**Comment by drone1** [comment]
Haven't tried it yet, but it looks like this is the actively maintained fork: https://github.com/VeliovGroup/flow-router
